assets/queries/k8s/cpu_requests/query.rego

```diff
@@ -1,67 +1,60 @@
 package Cx
 
+import data.generic.k8s as k8sLib
+
 CxPolicy[result] {
 	document := input.document[i]
-	specInfo := getSpecInfo(document)
+	specInfo := k8sLib.getSpecInfo(document)
 	metadata := document.metadata
 
-	containers := document.spec.containers
+	types := {"initContainers", "containers"}
+	containers := document.spec[types[x]]
 
 	object.get(containers[index].resources.requests, "cpu", "undefined") == "undefined"
 
 	result := {
 		"documentId": input.document[i].id,
 		"issueType": "MissingAttribute",
-		"searchKey": sprintf("metadata.name=%s.%s.containers.name=%s.resources.requests", [metadata.name, specInfo.path, containers[index].name]),
-		"keyExpectedValue": sprintf("%s.containers.name=%s.resources.requests does have CPU requests", [specInfo.path, containers[index].name]),
-		"keyActualValue": sprintf("%s.containers.name=%s.resources.requests doesn't have CPU requests", [specInfo.path, containers[index].name]),
+		"searchKey": sprintf("metadata.name=%s.%s.%s.name=%s.resources.requests", [metadata.name, specInfo.path, types[x], containers[index].name]),
+		"keyExpectedValue": sprintf("%s.%s.name=%s.resources.requests does have CPU requests", [specInfo.path, types[x], containers[index].name]),
+		"keyActualValue": sprintf("%s.%s.name=%s.resources.requests doesn't have CPU requests", [specInfo.path, types[x], containers[index].name]),
 	}
 }
 
 CxPolicy[result] {
 	document := input.document[i]
-	specInfo := getSpecInfo(document)
+	specInfo := k8sLib.getSpecInfo(document)
 	metadata := document.metadata
 
-	containers := document.spec.containers
+	types := {"initContainers", "containers"}
+	containers := document.spec[types[x]]
```

Comment:
this is also used in a lot of queries, can pass to the libraries as well

Reply:
this function would have to receive the document as input. If a document (or sample) has both containers and initContainers,  calling the function would produce the error: **eval_conflict_error: functions must not produce multiple outputs for same inputs**